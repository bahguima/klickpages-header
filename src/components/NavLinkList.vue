<template>
  <ul class="navbar__menu" id="kp-nav-header">
    <nav-link-item
      :url="topBarConfig.pages.url"
      :icon="'icon-kp-monitor'"
      :linkTitle="topBarConfig.pages.title"
    />
    <nav-link-item
      :url="topBarConfig.klicktest.url"
      :icon="'icon-kp-testab'"
      :linkTitle="topBarConfig.klicktest.title"
      :linkDescription="topBarConfig.klicktest.description"
      :enabled="topBarConfig.klicktest.enabled"
    />
    <nav-link-item
      :url="topBarConfig.klicktimer.url"
      :icon="'icon-kp-klicktimer'"
      :linkTitle="topBarConfig.klicktimer.title"
      :linkDescription="topBarConfig.klicktimer.description"
      :enabled="topBarConfig.klicktimer.enabled"
    />
    <nav-link-item
      :url="topBarConfig.klickbox.url"
      :icon="'icon-kp-pop-up'"
      :linkTitle="topBarConfig.klickbox.title"
      :linkDescription="topBarConfig.klickbox.description"
      :enabled="topBarConfig.klickbox.enabled"
    />
    <nav-link-item
      :url="topBarConfig.klickredirect.url"
      :icon="'icon-kp-link'"
      :linkTitle="topBarConfig.klickredirect.title"
      :linkDescription="topBarConfig.klickredirect.description"
    />
  </ul>
</template>

<script>

import { mapGetters } from 'vuex';
import NavLinkItem from './NavLinkItem.vue';

export default {
  name: 'NavLinkList',
  components: {
    NavLinkItem,
  },
  computed: mapGetters({
    topBarConfig: 'topBar/config',
  }),
};
</script>
