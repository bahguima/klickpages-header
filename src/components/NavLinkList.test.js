import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import NavLinkList from './NavLinkList.vue';

describe('components/NavLinkList.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const topBar = {
    namespaced: true,
    getters: {
      config: () => ({
        klicksend: {
          url: '',
          enabled: true,
        },
        home: {
          title: '',
          url: '',
        },
        klicktest: {
          title: '',
          description: '',
          url: '',
          enabled: true,
        },
        klicktimer: {
          title: '',
          description: '',
          url: '',
          enabled: true,
        },
        klickbox: {
          title: '',
          description: '',
          url: '',
          enabled: true,
        },
        klickredirect: {
          title: '',
          description: '',
          url: '',
        },
      }),
    },
  };

  describe('when component is mounted', () => {
    beforeAll(() => {
      const store = new Vuex.Store({ modules: { topBar } });
      wrapper = shallowMount(NavLinkList, { localVue, store });
    });

    it('should renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should contains navbar__menu class', () => {
      expect(wrapper.contains('.navbar__menu')).toBe(true);
    });

    afterAll(() => {
      wrapper.destroy();
      jest.clearAllMocks();
    });
  });
});
