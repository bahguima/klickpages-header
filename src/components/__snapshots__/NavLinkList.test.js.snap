// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavLinkList.vue when component is mounted should renders correctly 1`] = `
<ul
  class="navbar__menu"
  id="kp-nav-header"
>
  <nav-link-item-stub
    enabled="true"
    icon="icon-kp-monitor"
    linkdescription=""
    linktitle=""
    url=""
  />
   
  <nav-link-item-stub
    enabled="true"
    icon="icon-kp-testab"
    linkdescription=""
    linktitle=""
    url=""
  />
   
  <nav-link-item-stub
    enabled="true"
    icon="icon-kp-klicktimer"
    linkdescription=""
    linktitle=""
    url=""
  />
   
  <nav-link-item-stub
    enabled="true"
    icon="icon-kp-pop-up"
    linkdescription=""
    linktitle=""
    url=""
  />
   
  <nav-link-item-stub
    enabled="true"
    icon="icon-kp-link"
    linkdescription=""
    linktitle=""
    url=""
  />
</ul>
`;
